### PR TITLE
CORCI-1055 build: Setup for clamav mirror

### PR DESCRIPTION
**Pull request by JohnMalmberg**
Set up to use a local mirror of clamav

RPM-test-version: 1.3.0
Skip-checkpatch: true
Skip-build: true
Skip-func-test: true
Skip-unit-tests: true
Skip-rpm-centos-rpms: false

Signed-off-by: John E Malmberg <john.e.malmberg@intel.com>

### CODE REVIEW

File: ci/rpm/scan_daos_node.sh [hunk 1]
```diff
@@ -14,6 +14,18 @@ pushd "/var/tmp/${lmd_src}"
 sudo ./install.sh
   sudo ln -s /usr/local/maldetect/ /bin/maldet
 popd
+
+fc_conf="/etc/freshclam.conf"
+sudo chmod 766 "$fc_conf"
```
[the remark this answers is not in the document]
Comment (JohnMalmberg) [reply ci/rpm/scan_daos_node.sh:19]: The user it is running under is from the "others" match, so have to either open it up completely or use sudo to append the line.

I have not had good luck in having sudo appending lines to a file.

File: ci/rpm/scan_daos_node.sh [hunk 2]
```diff
@@ -14,6 +14,18 @@ pushd "/var/tmp/${lmd_src}"
...
+if ! grep -q 'ScriptedUpdates no' "$fc_conf"; then
+  printf "ScriptedUpdates no\n" >> "$fc_conf"
+fi
+: "${JOB_URL:=${JENKINS_URL}job/clamav_daily_update/}"
```
[the remark this answers is not in the document]
Comment (JohnMalmberg) [reply ci/rpm/scan_daos_node.sh:23]: Not looking for it in a comment.  Looking to see if VM is prebuilt with this edit already done.

File: ci/rpm/scan_daos_node.sh [hunk 3]
```diff
@@ -14,6 +14,18 @@ pushd "/var/tmp/${lmd_src}"
...
+if ! grep -q "$JENKINS_URL" "$fc_conf"; then
+  printf "PrivateMirror %s\n" \
+         "${JOB_URL}lastSuccessfulBuild/artifact/download/clam" >> "$fc_conf"
+fi
+sudo chmod 744 "$fc_conf"
```
[the remark this answers is not in the document]
Comment (JohnMalmberg) [reply ci/rpm/scan_daos_node.sh:28]: Again just checking to see if this edit already exists.

File: ci/rpm/scan_daos_node.sh [hunk 4]
```diff
@@ -14,6 +14,18 @@ pushd "/var/tmp/${lmd_src}"
 sudo ./install.sh
   sudo ln -s /usr/local/maldetect/ /bin/maldet
 popd
+
+fc_conf="/etc/freshclam.conf"
+sudo chmod 766 "$fc_conf"
+if ! grep -q 'ScriptedUpdates no' "$fc_conf"; then
+  printf "ScriptedUpdates no\n" >> "$fc_conf"
+fi
```
Comment (brianjmurrell) [inline ci/rpm/scan_daos_node.sh:22]:
```suggestion
if ! grep -q 'ScriptedUpdates no' "$fc_conf"; then
  sudo bash -c "echo \"ScriptedUpdates no\" >> \"$fc_conf\""
fi
```

`printf` is overkill for just adding a line with no formatting.

File: ci/rpm/scan_daos_node.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (brianjmurrell) [reply ci/rpm/scan_daos_node.sh:23]: But what if upstream adds a comment to the file with your match in it, as a way to further document possible options in the file?

File: ci/rpm/scan_daos_node.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (brianjmurrell) [reply ci/rpm/scan_daos_node.sh:28]: Yeah, this one is less critical since we are looking for a string with our URL in it.

Comment (brianjmurrell) [inline ci/rpm/scan_daos_node.sh:28]:
```suggestion

fc_conf="/etc/freshclam.conf"
if ! grep -q 'ScriptedUpdates no' "$fc_conf"; then
  sudo bash -c "echo \"ScriptedUpdates no\" >> \"$fc_conf\""
fi

: "${JOB_URL:=${JENKINS_URL}job/clamav_daily_update/}"
if ! grep -q "$JENKINS_URL" "$fc_conf"; then
  sudo bash -c "echo \"PrivateMirror ${JOB_URL}lastSuccessfulBuild/artifact/download/clam\" >> \"$fc_conf\""
fi
```

FWIW, `printf` is overkill for just adding a line with no formatting.

Comment (JohnMalmberg) [reply ci/rpm/scan_daos_node.sh:28]: Will be in the next push

File: Jenkinsfile [hunk 5]
```diff
@@ -1516,7 +1516,7 @@ pipeline {
         } // stage ('Test Report')
     } // stages
     post {
-        always {
+         always {
```
Comment (brianjmurrell) [inline Jenkinsfile:1519]: Looks like you introduced a 5th space in that indent.

File: ci/rpm/scan_daos_node.sh [hunk 6]
```diff
@@ -14,6 +14,16 @@ pushd "/var/tmp/${lmd_src}"
 sudo ./install.sh
   sudo ln -s /usr/local/maldetect/ /bin/maldet
```
Comment (brianjmurrell) [inline ci/rpm/scan_daos_node.sh:15]: Why is a directory being symlinked to `/bin`?  I can't think of any situation where a directory in `/bin` is useful

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
Do we know that this can be ignored?

```
[2021-04-02T00:42:38.638Z] Reading CVD header (daily.cld): Reading CVD header (daily.cvd): WARNING: remote_cvdhead: file not found: https://build.hpdd.intel.com/job/clamav_daily_update/lastSuccessfulBuild/artifact/download/clam/daily.cld
```

File: ci/rpm/scan_daos_node.sh (hunk 6 above)
Comment (JohnMalmberg) [reply ci/rpm/scan_daos_node.sh]: No idea why that was there.  Can not find the article that said to do it now.  Fixed in the current push.

File: ci/rpm/scan_daos_node.sh [hunk 7]
```diff
@@ -14,6 +14,16 @@ pushd "/var/tmp/${lmd_src}"
...
 popd
+
+fc_conf="/etc/freshclam.conf"
+if ! grep -q 'ScriptedUpdates no' "$fc_conf"; then
```
[the remark this answers is not in the document]
Comment (JohnMalmberg) [reply ci/rpm/scan_daos_node.sh]: That daily.cld warning can be ignored.   The mirror is no longer providing that file.

[the remark this answers is not in the document]
Comment (JohnMalmberg) [reply ci/rpm/scan_daos_node.sh]: Grep fixed to be inside of sudo.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

**Review by brianjmurrell — APPROVED** [review]

**Review by frostedcmos — APPROVED** [review]

**Pull request merged**